source/common/filesystem/kqueue/watcher_impl.cc

```diff
@@ -32,11 +32,13 @@ WatcherImpl::~WatcherImpl() {
   watches_.clear();
 }
 
-void WatcherImpl::addWatch(absl::string_view path, uint32_t events, Watcher::OnChangedCb cb) {
+absl::Status WatcherImpl::addWatch(absl::string_view path, uint32_t events,
```

Comment:
drive-by comment:
Seems this is calling the other `addWatch()` method that may also throw exceptions.
What I'm a bit concerned about is callers to this method, that now may assume that EnvoyExceptions won't occur, but then find out that it does.
IIUC this implies that the callers need to consider both error status and EnvoyException.

It may be better to either change the other `addWatch()` signature to return StatusOr (may end up changing manny other places in the code), or add a try-catch to the call of `addWatch()` in line 37, and convert an exception to an absl error.

Reply:
Thanks for calling this out.  I don't _think_ there's anything anywhere indicating that returning status means not throwing exceptions.  All of the 50 or so status conversion PRs thusfar have had the same property and will until we have an exception free build so I'm inclined to leave this as as I don't think adding try-catch locals in each conversion PR in the interim is plausible performance wise :-/

Reply:
Ack, thanks.